src/background/stats.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * Ghostery Browser Extension
+ * https://www.ghostery.com/
+ *
+ * Copyright 2017-present Ghostery GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+import AnonymousCommunication from '@whotracksme/webextension-packages/packages/anonymous-communication';
+import Storage from '@whotracksme/webextension-packages/packages/anonymous-communication/storage';
+import {
+  getTimeAsYYYYMMDDHH,
+  getTimeAsYYYYMMDD,
+  getTrustedUtcTime,
+} from '@whotracksme/webextension-packages/packages/anonymous-communication/timestamps';
+
+const HPN_URL = 'https://collector-hpn.ghostery.net';
+const CONFIG_URL = 'https://api.ghostery.net/api/v1/config';
+
+async function stats() {
+  const storage = new Storage('hpn-public-keys');
+  await storage.init();
+
+  const location = await fetch(CONFIG_URL)
+    .then((res) => res.json())
+    .then((res) => res.location);
+
+  const communication = new AnonymousCommunication({
+    config: {
+      COLLECTOR_DIRECT_URL: HPN_URL,
+      COLLECTOR_PROXY_URL: HPN_URL,
+    },
+    storage,
+  });
+
+  function send() {
+    const now = getTrustedUtcTime();
+    const ts = getTimeAsYYYYMMDD(now);
+    const t = getTimeAsYYYYMMDDHH(now);
+
+    communication.send({
+      action: 'wtm.alive',
+      ver: 1,
+      channel: 'safari',
+      ts,
+      payload: { t, ctry: location },
+    });
+  }
+
+  send();
+  setInterval(send, 1000 * 60 * 60 /* 1 hour */);
```

Comment:
I don't think we can use intervals safely in Manifest V3. Would be also better to send only on activity.

We might start with something like that, but I feel this part will have to change later.

Reply:
The PR is still in-progress (I checked it as a "draft"). I pushed it to the repository for a general overview. I think we need also to store the time of the sent message, to not send it again in an hour time window. 

However, what is wrong with setInterval? My intention was to call again `send` if the context still exists. If the browser kills the background process, it will be initialized again, and a message will be sent.

Reply:
Was just a general remark. My feeling is that we will need to move to a model, where some part of the system detects activity and then checks if a message needs to be sent. Extension startup is a natural place for that, so that is good. Most likely it is even sufficient for our purpose.

My comment is mostly that setTimeout is an API that might not do what you expect; for limited use cases (very brief timeouts), it will work, yet in the model that the extension ecosystems moves towards and also on Mobile browsers, I think it is a safe bet that the one hour timeout will never trigger.